From: Bruce Allan <ballan@redhat.com>
Subject: [RHEL5.1 PATCH] e1000: add support (device IDs) for HP Mezzanine cards
Date: Thu, 09 Aug 2007 11:56:00 -0700
Bugzilla: 251214
Message-Id: <46BB6340.5050604@redhat.com>
Changelog: [net] e1000: add support for HP Mezzanine cards


RHBZ#: 251214
------
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=251214

Description:
------------
Add PCI device IDs for 82571EB PRO/1000 Quad Port Gigabit Mezzanine Adapter.

Upstream Status:
----------------
This is not yet upstream but will be soon.

Test Status:
------------
This has been tested at Intel.

-- 
Bruce Allan
Intel LAN Access Division
2111 NE 25th Avenue
Hillsboro, OR  97124
503-712-5943

diff -urp -X dontdiff linux-2.6.18-38.el5/drivers/net/e1000/e1000_ethtool.c linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_ethtool.c
--- linux-2.6.18-38.el5/drivers/net/e1000/e1000_ethtool.c	2007-08-06 14:26:50.000000000 -0700
+++ linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_ethtool.c	2007-08-06 15:23:16.000000000 -0700
@@ -1711,6 +1711,7 @@ static int e1000_wol_exclusion(struct e1
 	case E1000_DEV_ID_82545EM_COPPER:
 	case E1000_DEV_ID_82546GB_QUAD_COPPER:
 	case E1000_DEV_ID_82546GB_PCIE:
+	case E1000_DEV_ID_82571EB_SERDES_QUAD:
 		/* these don't support WoL at all */
 		wol->supported = 0;
 		break;
diff -urp -X dontdiff linux-2.6.18-38.el5/drivers/net/e1000/e1000_hw.c linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_hw.c
--- linux-2.6.18-38.el5/drivers/net/e1000/e1000_hw.c	2007-08-06 14:26:53.000000000 -0700
+++ linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_hw.c	2007-08-06 15:23:16.000000000 -0700
@@ -384,6 +384,8 @@ e1000_set_mac_type(struct e1000_hw *hw)
 	case E1000_DEV_ID_82571EB_COPPER:
 	case E1000_DEV_ID_82571EB_FIBER:
 	case E1000_DEV_ID_82571EB_SERDES:
+	case E1000_DEV_ID_82571EB_SERDES_DUAL:
+	case E1000_DEV_ID_82571EB_SERDES_QUAD:
 	case E1000_DEV_ID_82571EB_QUAD_COPPER:
 	case E1000_DEV_ID_82571EB_QUAD_COPPER_LOWPROFILE:
 		hw->mac_type = e1000_82571;
@@ -485,6 +487,8 @@ e1000_set_media_type(struct e1000_hw *hw
     case E1000_DEV_ID_82545GM_SERDES:
     case E1000_DEV_ID_82546GB_SERDES:
     case E1000_DEV_ID_82571EB_SERDES:
+    case E1000_DEV_ID_82571EB_SERDES_DUAL:
+    case E1000_DEV_ID_82571EB_SERDES_QUAD:
     case E1000_DEV_ID_82572EI_SERDES:
     case E1000_DEV_ID_80003ES2LAN_SERDES_DPT:
         hw->media_type = e1000_media_type_internal_serdes;
diff -urp -X dontdiff linux-2.6.18-38.el5/drivers/net/e1000/e1000_hw.h linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_hw.h
--- linux-2.6.18-38.el5/drivers/net/e1000/e1000_hw.h	2007-08-06 14:26:53.000000000 -0700
+++ linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_hw.h	2007-08-06 15:23:16.000000000 -0700
@@ -473,6 +473,8 @@ int32_t e1000_check_phy_reset_block(stru
 #define E1000_DEV_ID_82571EB_SERDES      0x1060
 #define E1000_DEV_ID_82571EB_QUAD_COPPER 0x10A4
 #define E1000_DEV_ID_82571EB_QUAD_COPPER_LOWPROFILE  0x10BC
+#define E1000_DEV_ID_82571EB_SERDES_DUAL 0x10D9
+#define E1000_DEV_ID_82571EB_SERDES_QUAD 0x10DA
 #define E1000_DEV_ID_82572EI_COPPER      0x107D
 #define E1000_DEV_ID_82572EI_FIBER       0x107E
 #define E1000_DEV_ID_82572EI_SERDES      0x107F
diff -urp -X dontdiff linux-2.6.18-38.el5/drivers/net/e1000/e1000_main.c linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_main.c
--- linux-2.6.18-38.el5/drivers/net/e1000/e1000_main.c	2007-08-06 14:26:53.000000000 -0700
+++ linux-2.6.18-38.el5.bz251214/drivers/net/e1000/e1000_main.c	2007-08-06 15:23:16.000000000 -0700
@@ -107,6 +107,8 @@ static struct pci_device_id e1000_pci_tb
 	INTEL_E1000_ETHERNET_DEVICE(0x10BC),
 	INTEL_E1000_ETHERNET_DEVICE(0x10C4),
 	INTEL_E1000_ETHERNET_DEVICE(0x10C5),
+	INTEL_E1000_ETHERNET_DEVICE(0x10D9),
+	INTEL_E1000_ETHERNET_DEVICE(0x10DA),
 	/* required last entry */
 	{0,}
 };

